### PR TITLE
OBPIH-4770 Create new product list

### PR DESCRIPTION
**Pull request by kchelstowski**
I also found in the mean time some minor issues on PO list and changed some state names on invoiceList which were copied from PO list and not changed by mistake.

### CODE REVIEW

File: src/js/components/products/ProductsList.jsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import ProductsListHeader from 'components/products/ProductsListHeader';
+import ProductsListTable from 'components/products/ProductsListTable';
+
+
+const ProductsList = (props) => {
+  // Filter params are stored here, to be able to pass them to table component
+  const [filterParams, setFilterParams] = useState({});
+
+  useEffect(() => {
+    props.fetchTranslations(props.locale, 'purchaseOrder');
```
Comment (drodzewicz) [inline src/js/components/products/ProductsList.jsx:17]: wrong translation fetched
change to
`props.fetchTranslations(props.locale, 'productsList');`

File: src/js/components/products/ProductsListFilters.jsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    getDynamicAttr: ({ categories }) => ({
+      options: categories,
+    }),
+  },
+  includeCategoryChildren: {
+    type: CheckboxField,
+    label: 'react.productsList.includeSubcategories.label',
+    defaultMessage: 'Include all products in all subcategories',
+    attributes: {
+      filterElement: true,
```
Comment (drodzewicz) [inline src/js/components/products/ProductsListFilters.jsx:30]: what does the attribute `filterElement: true` do for  `CheckboxField` ?

**Review by drodzewicz — CHANGES_REQUESTED** [review]

File: src/js/components/products/ProductsListFilters.jsx (hunk 2 above)
Comment (kchelstowski) [reply src/js/components/products/ProductsListFilters.jsx:30]: it is added to filter form elements not to have to play around with position styles included in `form-utils`, so it is just to position filter elements, nothing functional.

Comment (drodzewicz) [reply src/js/components/products/ProductsListFilters.jsx:30]: that's right I remember now. The naming implies that it has something to do with filtering, kind of misleading

**Review by drodzewicz — APPROVED** [review]

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**